presto-client/src/client.ts

```diff
@@ -91,11 +94,25 @@ export class PrestoClient {
     do {
       const response = await this.request({ method: 'GET', url: nextUri })
 
+      // Server is overloaded, wait a bit
+      if (response.status === 503) {
+        await this.delay(this.retryInterval)
+        continue
+      }
+
       if (response.status !== 200) {
         throw new Error(`Query failed: ${JSON.stringify(await response.text())}`)
       }
 
       const prestoResponse = (await response.json()) as PrestoResponse
+      if (!prestoResponse) {
+        throw new Error(`Query failed with an empty response from the server.`)
+      }
+
+      const { error } = prestoResponse
+      if (error) {
```

Comment:
if error is only used here maybe it is better to 
```suggestion
      if (prestoResponse.error) {
```